packages/design-system-react/src/components/text/Text.tsx

```diff
@@ -13,10 +14,11 @@ export const Text: React.FC<TextProps> = ({
   textAlign,
   overflowWrap,
   ellipsis,
-  as,
+  asChild,
   color = TextColor.TextDefault,
+  style,
 }) => {
-  const Tag = as || TEXT_DEFAULT_TAG_MAP[variant];
+  const Component = asChild ? Slot : TEXT_DEFAULT_TAG_MAP[variant];
```

Comment:
If `asChild` is try we will use the Radix slot component to merge props otherwise use the default tag from variant